spec/models/reports/view_spec.rb

```diff
@@ -31,25 +31,52 @@ def table_descriptions_sql(table_name)
     end
   end
 
+  describe ".fetch_refresh_month_for_date" do
+    it "returns current and previous month when date is odd" do
+      date_of_refresh = Date.new(2023, 6, 15)
+      expect(described_class.fetch_refresh_month_for_date(date_of_refresh)).to eq(Date.new(2023, 5, 1))
+    end
+
+    it "returns current month and month offset when date is even" do
+      date_of_refresh = Date.new(2023, 6, 16)
+      expect(described_class.fetch_refresh_month_for_date(date_of_refresh)).to eq(Date.new(2022, 9, 1))
+    end
+
+    it "handles month transitions correctly for odd days" do
+      date_of_refresh = Date.new(2023, 1, 1)
+      expect(described_class.fetch_refresh_month_for_date(date_of_refresh)).to eq(Date.new(2022, 12, 1))
+    end
+  end
+
   describe ".get_refresh_months" do
-    it "returns current and previous month when current day is odd" do
-      travel_to Date.new(2023, 6, 15) do
-        expected_months = [Date.new(2023, 6, 1), Date.new(2023, 5, 1)]
-        expect(described_class.get_refresh_months).to eq(expected_months)
+    context "when refresh frequency is not set" do
+      before { ENV.delete("REPORTING_REFRESH_FREQUENCY") }
+
+      it "defaults to daily logic" do
+        travel_to(Date.new(2025, 9, 24)) do
```

Comment:
Hold on a second! This is possible? That's amazing. Means we can do without the Timecop gem.

https://github.com/simpledotorg/simple-server/blob/cd2124fa67eee59c519f461fc5e75e7620577e41/Gemfile#L87